scripts/process_new_format.py

```
@@ -47,40 +48,48 @@
     UPDATE = auto()
     AS_IF = auto()
     AUTOMATIC_SUCCESS_FAILURE = auto()
     AUTOMATIC_SUCCESS_FAILURE_AUTOMATIC_EVASION = auto()
 
 
-class Ruling(BaseModel):
-    class Config:
-        arbitrary_types_allowed = True
+class Provenance(BaseModel):
+    source_type: str  # E.g., "arkhamdb_faq", "official_faq_pdf", "email_ruling"
+    source_name: Optional[str] = None  # E.g., "FAQ v1.7", "Matt Newman Email"
```

Ruff (0.11.9)

`datetime.datetime.utcnow()` used

(DTZ003)